src/category_theory/sites/cover_preserving.lean

```diff
@@ -118,6 +125,43 @@ end
 
 omit h hG
 
+open limits.walking_cospan
+
+lemma compatible_preserving_of_flat {C : Type u₁} [category.{v₁} C] {D : Type u₁} [category.{v₁} D]
+  (K : grothendieck_topology D) (G : C ⥤ D) [representably_flat G] : compatible_preserving K G :=
+begin
+  constructor,
+  intros ℱ Z T x hx Y₁ Y₂ X f₁ f₂ g₁ g₂ hg₁ hg₂ e,
+
+  /- First, `f₁` and `f₂` forms a cone over `cospan g₁ g₂ ⋙ u`. -/
+  let c : cone (cospan g₁ g₂ ⋙ G) :=
+    (cones.postcompose (diagram_iso_cospan (cospan g₁ g₂ ⋙ G)).inv).obj
+      (pullback_cone.mk f₁ f₂ e),
+
+  /-
+  This can then be viewed as a cospan of structured arrows, and we may obtain an arbitrary cone
+  over it since `structured_arrow W u` is cofiltered.
+  Then, it suffices to prove that it is compatible when restricted onto `u(c'.X.right)`.
+  -/
+  let c' := is_cofiltered.cone (structured_arrow_cone.to_diagram c ⋙ structured_arrow.pre _ _ _),
+  have eq₁ : f₁ = (c'.X.hom ≫ G.map (c'.π.app left).right) ≫ eq_to_hom (by simp),
+  { erw ← (c'.π.app left).w, dsimp, simp },
+  have eq₂ : f₂ = (c'.X.hom ≫ G.map (c'.π.app right).right) ≫ eq_to_hom (by simp),
+  { erw ← (c'.π.app right).w, dsimp, simp },
+  conv_lhs { rw eq₁ },
+  conv_rhs { rw eq₂ },
+  simp only [op_comp, functor.map_comp, types_comp_apply, eq_to_hom_op, eq_to_hom_map],
+  congr' 1,
+
+  /-
+  Now, since everything now falls in the image of `u`,
```

Comment:
Nitpick: you are using "now" twice on the same line. I think one of them can be dropped.